Library/Homebrew/cmd/update.sh

```diff
@@ -35,7 +35,9 @@ git_init_if_necessary() {
     git config --bool core.autocrlf false
     git config remote.origin.url "$BREW_OFFICIAL_REMOTE"
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
-    git fetch --force --depth=1 origin refs/heads/master:refs/remotes/origin/master
+    latest_tag="$(git ls-remote --tags --refs -q origin | tail -n1)"
+    latest_tag_commit="${latest_tag%$'\t'*}"
+    git fetch --force origin --shallow-since "$latest_tag_commit"
```

Comment:
Minor point of style, I prefer separating with `=` rather than space. I'm not sure whether Homebrew has a style guide addressing this.
````
--shallow-since="$latest_tag_commit"
```